Examples/TodoApp/TodoApp/src/data/Task.swift

```diff
@@ -1,6 +1,6 @@
 import Foundation
 
-struct Task: Equatable {
+struct Task: Equatable, Codable {
     let id: String
     var details: TaskDetails
 
```

Comment:
Not specific to the changes in this PR, but I noticed that creating new tasks without assigning an `id` uses the empty string default, which causes multiple creates to overwrite each other.

Instead of putting the responsibility on the caller, I think we should change the `init` signature to create a UUID value unless a specific id is passed in by the caller.

```
init(id: String? = nil, details: TaskDetails) {
    self.id = id ?? UUID().uuidString
    self.details = details
    isCompleted = false
    isActive = false
}
```

What do you think?

Reply:
👍